src/file_pipe_log.rs

```diff
@@ -794,6 +811,32 @@ mod tests {
         assert!(parse_file_name(invalid_file_name).is_err());
     }
 
+    #[test]
+    fn test_dir_lock() {
+        let dir = Builder::new().prefix("test_lock").tempdir().unwrap();
+        let path = dir.path().to_str().unwrap();
+
+        let mut cfg = Config::default();
+        cfg.dir = path.to_owned();
+        cfg.bytes_per_sync = ReadableSize(1024u64);
+        cfg.target_file_size = ReadableSize(32 * 1024 as u64);
+
+        let r1 = FilePipeLog::open::<BlackholeSequentialReplayMachine>(
+            &cfg,
+            Arc::new(DefaultFileBuilder {}),
+            vec![],
+        );
+
+        assert_eq!(r1.is_ok(), true);
+        // Only one thread can hold file lock
+        let r2 = FilePipeLog::open::<BlackholeSequentialReplayMachine>(
+            &cfg,
+            Arc::new(DefaultFileBuilder {}),
+            vec![],
+        );
+        assert_eq!(r2.is_err(), true);
```

Comment:
Could you verify the error message is `failed to create lock` or `maybe another instance is using this directory`? I checked that file lock is a re-entrant lock, which means the second `FilePipeLog` can lock the file just fine.

Reply:
OK, i will more the test more accurate